src/generic/graphicc.cpp

```diff
@@ -1608,6 +1608,21 @@ wxCairoBitmapData::wxCairoBitmapData( wxGraphicsRenderer* renderer, cairo_surfac
     m_buffer = nullptr;
 }
 
+namespace
+{
+#ifdef wxHAS_RAW_BITMAP
+// Return true if the pixel is masked under this platform
+inline bool IsMasked(wxNativePixelData::Iterator& p)
+{
+#ifndef __WXQT__
+    return p.Red()+p.Green()+p.Blue() == 0;
+#else
+    return p.Red()+p.Green()+p.Blue() != 0;
```

Comment:
`wxMask` documentation (https://docs.wxwidgets.org/trunk/classwx_mask.html) states that "masked area is black and the unmasked area is white" and `wxQT` should conform this convention.

Reply:
From the user's POV, wxQt conforms to the documentation. that is, applying the same mask on the same bitmap should give the same result on all platforms.

For the implementation, one should know that Qt applies the mask in opposite way to what wx documentation says. IOW, masked area is white and the unmasked area is black! and you can see how Qt actually applies the mask by the following patch:
```
diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 5f32257c6b..7170fd1e19 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -26,6 +26,9 @@
     #include "wx/wx.h"
 #endif
 
+#include <QtGui/QPixmap>
+#include <QtGui/QBitmap>
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -188,16 +191,14 @@ void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 
 void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
-    wxMessageBox(wxString::Format
-                 (
-                    "Welcome to %s!\n"
-                    "\n"
-                    "This is the minimal wxWidgets sample\n"
-                    "running under %s.",
-                    wxVERSION_STRING,
-                    wxGetOsDescription()
-                 ),
-                 "About wxWidgets minimal sample",
-                 wxOK | wxICON_INFORMATION,
-                 this);
+    QPixmap pix;
+    pix.load("colour.bmp");
+    QBitmap mask;
+    mask.load("mask.bmp");
+
+    if ( !pix.isNull() && !mask.isNull() )
+    {
+        pix.setMask(mask);
+        pix.save("colour_with_mask.bmp");
+    }
 }
```
colour.bmp:
![colour](https://github.com/wxWidgets/wxWidgets/assets/7704771/cd00acb2-fe76-4775-8034-069acd1b7d50)

mask.bmp:
![mask](https://github.com/wxWidgets/wxWidgets/assets/7704771/d9d545d2-9bcf-4ac0-b97a-81dbc1e7ee21)

colour_with_mask.bmp (expected):
![colour_with_mask_expected](https://github.com/wxWidgets/wxWidgets/assets/7704771/79fa4b84-edd4-4109-8ac5-1e716d00f7a0)

colour_with_mask.bmp(actual result!):
![colour_with_mask_result](https://github.com/wxWidgets/wxWidgets/assets/7704771/6079e793-561e-4d21-8156-83278ba0d153)

So I don't know how to make things more transparent under wxQt without complicating wxBitmap implementation further and making it inefficient at best!

Reply:
I'm not proficient with QT but it seems that in `wxImage ConvertImage(QImage qtImage)` when you convert 1bpp mask to RGB with
`if ( qtImage.depth() == 1 )
      qtImage = qtImage.convertToFormat(QImage::Format_RGB32);
`
you could also invert it using `QImage::invertPixels()`.
I think it shouldn't degrade performance very much (`QImage::converttoFormat()` itself is probably expensive).
With inverting  QImage internally you should be able to handle wx-style mask.